### ISSUE TITLE
The visual graphs in the examples have a colour based legend 

### ISSUE THREAD

**Issue by DavidMacDonald** [wet-boew/wet-boew#5994]
There is a colour based legend on the pie charts. http://wet-boew.github.io/v4.0-ci/demos/charts/custom-en.html 
There is a fallback table but this is insufficient for sighted people who are colour blind... I've attached an image of what should be provided to ensure adequate accessibility for people who are colour blind. The items should be right on the image.. or associated by lines when it can't fit...

![2014-08-31 6-18-01 pm](https://cloud.githubusercontent.com/assets/2365411/4103427/48232514-315d-11e4-9a77-d2ad410365f1.png)

**Comment by hsrudnicki** [comment]
Can be this issue closed?

**Comment by pjackson28** [comment]
@duboisp Has this been addressed?

**Comment by duboisp** [comment]
Hi @DavidMacDonald 

I know the following may not address completely your issue, but at least it will provide more guidance on how to create a charts that could be closer to your proposal.

Labels can be added to pie charts by using the ```data-flot``` data attribute, configured as per the [Flot pie chart documentation](http://www.flotcharts.org/flot/examples/series-pie/index.html), on the table elements.

But currently, from a quick and easy perspective and by using Flot to render the charts, the position of each labels are consistent for each slice in the pie charts. Line cann't be draw between label and their associated slice. And it's possible to include the legend on the pie charts.

## Adding label to the chart

### Example 1 - Show the defaults labels on each quarter
_The following example is to show how to do it. Please take in consideration, in order to fit your need, it may require to use different value._
```xml
<table class="wb-charts wb-charts-pie table"
	data-flot='{
		"series": {
			"pie": {
				"label": {
					"show": true
				}
			}
		}
	}'
>
```
![example1](https://cloud.githubusercontent.com/assets/1676030/8390397/f730aebc-1c5f-11e5-9fc6-96ddfb84902c.PNG)

### Example 2 - Ajusting the position of the labels
_The following example is to show how to do it. Please take in consideration, in order to fit your need, it may require to use different value._
```xml
<table class="wb-charts wb-charts-pie table"
	data-flot='{
		"series": { 
			"pie": { 
				"radius": 1, 
				"label": { 
					"radius": 1,
					"show": true 
				} 
			} 
		} 
	}'
>
```
![example2](https://cloud.githubusercontent.com/assets/1676030/8390399/31fd85f6-1c60-11e5-917a-ec16e3e24d09.PNG)

You can see that I play with the value of ```radius``` [with a range from 0 to 1] on the pie chart and on the label for adjusting the label position. But currently that position is generalized to each quarter where you can have labels that are superposing like this:

![example2a](https://cloud.githubusercontent.com/assets/1676030/8390407/f07a3358-1c60-11e5-9771-9b87dc3ab229.PNG)

## Overlapping labels

The Flot pie chart plugin provide 2 solution but none of them do a nice line that link the label with the slice. 
***WET Developer note:*** A solution that will do a line between the label and the slice could require to update the Flot pie chart plugin, or use/create a different Flot plugin or use another charting plugin other than Flot that support it.

### Label overlapping solution using ```threshold```
Slice where their value are equal or less to the threshold wont have a label.
_The following example is to show how to do it. Please take in consideration, in order to fit your need, it may require to use different value._
```xml
<table class="wb-charts wb-charts-pie table"
	data-flot='{
		"series": { 
			"pie": { 
				"radius": 1, 
				"label": { 
					"radius": 1, 
					"show": true,
					"threshold": 0.05
				} 
			} 
		} 
	}'
>
```
![example3](https://cloud.githubusercontent.com/assets/1676030/8390434/9335efb4-1c62-11e5-9622-165688fd296a.PNG)

### Label overlapping solution using ```combine```
Slice under a threshold will be combined together and will use the specified text label in the configuration.
_The following example is to show how to do it. Please take in consideration, in order to fit your need, it may require to use different value._
```xml
<table class="wb-charts wb-charts-pie table"
	data-flot='{
		"series": { 
			"pie": { 
				"radius": 1, 
				"label": { 
					"radius": 1, 
					"show": true,
					"threshold": 0.05
				},
				"combine": {
					"threshold": 0.05,
					"color": "#555",
					"label": "Combined slice"
				}
			} 
		} 
	}'
>
```

![example4](https://cloud.githubusercontent.com/assets/1676030/8390450/d81753c4-1c63-11e5-99dc-049a4b1a9fdf.PNG)

## Customizing the label content

In your example, you used the legend content. It is possible to achieve the same by setting a custom label ```formatter``` function.

The method to do that in WET and currently only supported for the charts require to create a "[Custom preset](http://wet-boew.github.io/v4.0-ci/demos/charts/custom-en.html)" then [declaring your custom preset function](http://wet-boew.github.io/v4.0-ci/docs/ref/charts/charts-en.html#functionpreset) and then use the new preset to see the magic. Read the _Tips and trick_ section in the [charts documentation](http://localhost/github/wet-boew-chart-graph/dist/unmin/docs/ref/charts/charts-en.html) for more details and more examples.

_The following example is to show how to do it. Please take in consideration, in order to fit your need, it may require to use different value._

Here an example that only show the legend on the slice:
```xml
<script>
	window[ "wb-charts" ] = {
		flot: {
			slicelegend: {
				fn: {
					"/series/pie/label/formatter": function( label, series ) {
						return label;
					}
				}
			} 
		} 
	};
</script>
	
<table class="wb-charts wb-charts-pie table wb-charts-slicelegend"
	data-flot='{
		"series": { 
			"pie": { 
				"radius": 1, 
				"label": { 
					"radius": 1, 
					"show": true,
					"threshold": 0.05
				},
				"combine": {
					"threshold": 0.05,
					"color": "#555",
					"label": "Combined slice"
				}
			} 
		} 
	}'
>
```

![example5](https://cloud.githubusercontent.com/assets/1676030/8390556/ad4c4506-1c67-11e5-880e-b600e7c0d4f9.PNG)

### Merging the configuration

Just to have cleaner HTML code, it is possible to merge the inline table data configuration and make it a simple preset. 

_This **is optional** and probably off context for the issue discussed in this tickets, but it could be interesting for a developer to know how to re-use preset configuration on other table to generate similar charts._

Here the code:
```xml
<script>
	window[ "wb-charts" ] = {
		flot: {
			slicelegend: {
				series: { 
					pie: { 
						radius: 1, 
						label: { 
							radius: 1, 
							show: true,
							threshold: 0.05
						},
						combine: {
							threshold: 0.05,
							color: "#555",
							label: "Combined slice"
						}
					}
				},
				fn: {
					"/series/pie/label/formatter": function( label, series ) {
						return label;
					}
				}
			} 
		} 
	};
</script>
	
<table class="wb-charts wb-charts-pie table wb-charts-slicelegend">
```
It's possible to simplify the code further by specifying a baseline preset to use. 
```xml
<script>
	window[ "wb-charts" ] = {
		flot: {
			slicelegend: {
				base: "pie",
				series: { 
					pie: { 
						radius: 1, 
						label: { 
							radius: 1, 
							show: true,
							threshold: 0.05
						},
						combine: {
							threshold: 0.05,
							color: "#555",
							label: "Combined slice"
						}
					}
				},
				fn: {
					"/series/pie/label/formatter": function( label, series ) {
						return label;
					}
				}
			} 
		} 
	};
</script>
	
<table class="wb-charts wb-charts-slicelegend table">
```

The both above example will produce this charts

![example6](https://cloud.githubusercontent.com/assets/1676030/8390576/650edb44-1c69-11e5-8cb6-d9b5d4737e80.PNG)

----
@pjackson28, @thomasgohard Shoud I create a similar basic preset directly in the chart plugin? If so... What could be the name of the preset? _no name, just as the pie charts default?_ What overlapping label avoiding strategy should be used? _Combine?_ What could be a reasonable threshold? _5%?_

----
Cheers

:smile:

**Comment by hsrudnicki** [comment]
@DavidMacDonald and others: Please update this issue's status whether this issue can be closed as resolved/fixed or otherwise? 
In case if you don't respond/update this issue's status soon, then this issue will be made "Going Stale" and closed very soon!

**Comment by hsrudnicki** [comment]
Due to no further discussion/status updates, this issue has been closed.

**Comment by bsouster** [comment]
To do: 
* [ ] Integrate the JS code to the chart plugins
* [ ] create working examples
